tests/bench/fixtures/fixtures.go

```diff
@@ -28,18 +29,41 @@ var (
 	}
 )
 
+type Options struct {
+	directives map[string]map[string][]string
+}
+
+func NewOptions() *Options {
+	return &Options{
+		directives: make(map[string]map[string][]string),
+	}
+}
+
+func (o *Options) WithFieldDirective(typeName, field, directive string) *Options {
+	if o.directives == nil {
+		o.directives = make(map[string]map[string][]string)
+	}
+	if o.directives[typeName] == nil {
+		o.directives[typeName] = make(map[string][]string)
+	}
+	o.directives[typeName][field] = append(o.directives[typeName][field], directive)
+	return o
+}
+
 type Generator struct {
 	ctx context.Context
 
-	schema string
-	types  []any
+	schema  string
+	types   []any
+	options *Options
 }
 
-func ForSchema(ctx context.Context, schemaName string) Generator {
+func ForSchema(ctx context.Context, schemaName string, options *Options) Generator {
```

Comment:
suggestion: I think this would be much nicer if `options` was optional, instead of forcing the default to call it with an explicit nil param (there are lots of `ForSchema(foo, bar, nil)` calls now).  I'd also suggest changing `Options` to something array based, it would problably fit better.

```
type Options = func(Generator) Generator // or similar, perhaps you'd want an Options struct on Generator, but keep the func array as a param to ForSchema
func ForSchema(ctx context.Context, schemaName string, options ...Options)
```